assets/section-main-product.css

```diff
@@ -233,6 +233,10 @@ fieldset.product-form__input .form__label {
   color: rgb(var(--color-background));
 }
 
+.product-form__input input[type='radio']:not(.disabled) + label > .visually-hidden {
+  display: none;
```

Comment:
I'm not sure whether **having <span> with string for all labels and showing them for only unavailable variants** is better than **adding this string to only unavailable variants using JS**. But your approach does its job and reduces using JS. Works for me for sure.